src/main/java/duke/Storage/Storage.java

```diff
@@ -0,0 +1,130 @@
+import java.io.FileWriter;
+import java.io.IOException;
+import java.nio.file.Files;
+import java.nio.file.Paths;
+import java.util.ArrayList;
+
+/**
+ * CLass responsible for interacting with List of task storing and loading from storage.
+ */
+public class Storage {
+    private final String filepath;
+    private final String filename;
+
+    /**
+     * Constructor for the Storage class.
+     */
+    public Storage() {
+        this.filepath = System.getProperty("user.dir") + File.separator + "data";
+        this.filename = "duke.Storage.txt";
+        try {
+            if (!Files.exists(Paths.get(filepath))) {
+                Files.createDirectories(Paths.get(filepath));
+            }
+            if (!Files.exists(Paths.get(filepath + File.separator + filename))) {
+                File file = new File(filepath + File.separator + filename);
+                boolean result = file.createNewFile();
+                if (!result) {
+                    System.out.println("Unable to create file");
+                }
+            }
+        } catch (IOException e) {
+            System.out.println("Unable to create path/file");
+        }
+    }
+
+    /**
+     * Reads from the storage and returns the stored list of tasks.
+     *
+     * @return an array list containing the stored list of tasks
+     */
+    public ArrayList<Task> readFromStorage() {
+        ArrayList<Task> listOfTasks = new ArrayList<>();
+
+        String st;
+        Task task = new Task("null");
+        try {
+            BufferedReader reader = new BufferedReader(new FileReader(this.filepath
+                                                        + File.separator + this.filename));
+            st = reader.readLine();
+            while (st != null) {
+                String[] compactTask = st.split("-");
+                switch (compactTask[0]) {
+                case "T":
+                    task = new ToDos(compactTask[2]);
+                    if (compactTask[1].equals("1")) {
+                        task.mark();
+                    }
+                    break;
+                case "D":
+                    task = new Deadlines(compactTask[2], compactTask[3]);
+                    if (compactTask[1].equals("1")) {
+                        task.mark();
+                    }
+                    break;
+                case "E":
+                    task = new Event(compactTask[2], compactTask[3]);
+                    if (compactTask[1].equals("1")) {
+                        task.mark();
+                    }
+                    break;
+                default:
+                    // Do nothing
+                }
+                listOfTasks.add(task);
+                st = reader.readLine();
+            }
+            reader.close();
+        } catch (DukeException ex) {
+            System.out.println("IOException while trying to read from file");
+        } catch (IOException ex) {
+            System.out.println("Error in storage format");
```

Comment:
Since UI (not sure if GUI serves the same purpose as I have not touched level 10) is responsible for interactions with users, would you like to consider leaving all the `System.out.println()` to ui? e.g. replace here with `ui.showError(msg)` so that all messages displayed are directly from ui 😜